src/robot/libraries/Collections.py

```diff
@@ -148,21 +152,19 @@ def remove_from_list(self, list_, index):
         | ${x} = 'a'
         | ${L2} = ['b']
         """
-        self._validate_list(list_)
         try:
-            return list_.pop(self._index_to_int(index))
+            return list_.pop(index)
         except IndexError:
             self._index_error(list_, index)
 
-    def remove_duplicates(self, list_):
+    def remove_duplicates(self, list_: ListLike) -> list:
```

Comment:
I think this should be `MutableSequence` or `Sequence`. Implementation doesn't require that, but `Mapping` or `set` cannot even have duplicates. For consistency reasons I think modifying keywords should only accept sequences, validation in general work also with others.